### PR TITLE
Release/3.0.2

### PR DESCRIPTION
**Pull request by dkijania**
Merge release 3.0.2 to master

### CODE REVIEW

**Comment by dkijania** [comment]
!ci-build-me

**Comment by dkijania** [comment]
!ci-nightly-me

**Comment by dkijania** [comment]
Nightly: https://buildkite.com/o-1-labs-2/mina-end-to-end-nightlies/builds/2763

**Review by SanabriaRusso — APPROVED** [review]

**Comment by mrmr1993** [comment]
!approved-for-mainnet

File: src/lib/crypto/kimchi_bindings/stubs/Cargo.toml [hunk 1]
```diff
@@ -13,22 +13,22 @@ crate-type = ["lib", "staticlib"]
...
+rand = "0.8.5"
+rayon = "1.5.0"
 serde = "1.0.130"
-serde_json = "1.0.91"
+serde_json = "1.0.103"
 sprs = { version = "0.11.0", features = ["multi_thread"] }
 once_cell = "1.10.0"
 
 # arkworks
-ark-ff = { version = "0.3.0", features = ["parallel"] }
```
Comment (dannywillems) [inline src/lib/crypto/kimchi_bindings/stubs/Cargo.toml:28]: Looks that it includes the bump up of arkworks. This should be added to the changelog for the release IMO.

File: src/lib/crypto/kimchi_bindings/stubs/src/lagrange_basis.rs [hunk 2]
```diff
@@ -0,0 +1,191 @@
+use ark_ec::AffineRepr;
```
Comment (dannywillems) [inline src/lib/crypto/kimchi_bindings/stubs/src/lagrange_basis.rs:1]: Should we add this to the CHANGELOG?

File: src/lib/crypto/kimchi_bindings/wasm/Cargo.toml [hunk 3]
```diff
@@ -53,17 +53,18 @@ rayon = { version = "1" }
 rmp-serde = "1.0.0"
 sprs = { version = "0.11.0" }
 spmc = { version = "0.3.0" }
-serde = "1.0.130"
-serde_json = "1.0"
```
Comment (dannywillems) [inline src/lib/crypto/kimchi_bindings/wasm/Cargo.toml:57]: Is it worth adding the bump up of the deps in the CHANGELOG?

File: src/lib/crypto/kimchi_bindings/wasm/src/arkworks/bigint_256.rs [hunk 4]
```diff
@@ -29,15 +29,18 @@ impl FromWasmAbi for WasmBigInteger256 {
     type Abi = <Vec<u8> as FromWasmAbi>::Abi;
     unsafe fn from_abi(js: Self::Abi) -> Self {
         let bytes: Vec<u8> = FromWasmAbi::from_abi(js);
-        WasmBigInteger256(BigInteger256(FromBytes::read(bytes.as_slice()).unwrap()))
+        // TODO this used FromBytes before arkworks 0.4.2, check serialization is consistent after update
```
Comment (dannywillems) [inline src/lib/crypto/kimchi_bindings/wasm/src/arkworks/bigint_256.rs:32]: cc @volhovm for this TODO. I guess it is correct if we do have regression tests?

**Review by dannywillems — APPROVED** [review]
I am approving the changes for src/lib/crypto and src/lib/pickles, with small comments:
- I don't see @querolita changes regarding zk_rows and num_chunks. Expected?
- @volhovm work is important IMO regarding the bump up of arkworks. I would add it in the CHANGELOG.
- many dependencies have been updated in the Rust code. I wish we can add this to the changelog every time we do it.

File: src/lib/crypto/kimchi_bindings/stubs/Cargo.toml (hunk 1 above)
Comment (dannywillems) [reply src/lib/crypto/kimchi_bindings/stubs/Cargo.toml:28]: Ignore. It is 3.0.2 into master. Not 3.0.3

File: src/lib/crypto/kimchi_bindings/stubs/src/lagrange_basis.rs (hunk 2 above)
Comment (dannywillems) [reply src/lib/crypto/kimchi_bindings/stubs/src/lagrange_basis.rs:1]: Ignore. It is 3.0.2 into master. Not 3.0.3

File: src/lib/crypto/kimchi_bindings/wasm/src/arkworks/bigint_256.rs (hunk 4 above)
Comment (dannywillems) [reply src/lib/crypto/kimchi_bindings/wasm/src/arkworks/bigint_256.rs:32]: Ignore. It is 3.0.2 into master. Not 3.0.3

**Comment by dannywillems** [comment]
> I am approving the changes for src/lib/crypto and src/lib/pickles, with small comments:
> 
>     * I don't see @querolita changes regarding zk_rows and num_chunks. Expected?
> 
>     * @volhovm work is important IMO regarding the bump up of arkworks. I would add it in the CHANGELOG.
> 
>     * many dependencies have been updated in the Rust code. I wish we can add this to the changelog every time we do it.

Ignore. It is 3.0.2 into master. Not 3.0.3

File: src/lib/crypto/kimchi_bindings/stubs/Cargo.toml (hunk 1 above)
Comment (volhovm) [reply src/lib/crypto/kimchi_bindings/stubs/Cargo.toml:28]: Why ignore? This is 3.0.2, the current release, and it has arkworks bump. So indeed we should include it I think.

Comment (volhovm) [reply src/lib/crypto/kimchi_bindings/stubs/Cargo.toml:28]: Yeah, 3.0.2 changelog is already finalized. We were talking about 3.0.3 indeed. Closing this :)

**Comment by georgeee** [comment]
https://gist.github.com/georgeee/a864f0314789caffa3c5f535244bf81b

**Review by deepthiskumar — APPROVED** [review]

**Review by Trivo25 — APPROVED** [review]

**Pull request merged**